.azdo/pipelines/generate-package.yml

```diff
@@ -0,0 +1,101 @@
+trigger:
+  tags:
+    include:
+    - '*'
+
+variables:
+  - name: VERSION-ID
+    value: $[replace(variables['BUILD.SOURCEBRANCHNAME'], 'v', '')]
+  - name: DEPENDENCY-PACK-NAME
+    value: "dependency-pack_$(VERSION-ID).deb"
+  - name: DEPENDENCY-PACK-PINNING-NAME
+    value: "dependency-pack-pinning_$(VERSION-ID).deb"
+
+stages:
+- stage: DEV
+  pool:
+    vmImage: ubuntu-latest
+
+  variables:
+  - group: DEV
+
+  jobs:
+  - job: generate_packages_for_dev
+    displayName: Generate packages for DEV environment
+    steps:
+    - script: src/build-package.sh 'package-history' 'output' "$(VERSION-ID)"
+      displayName: 'Generate packages locally'
+
+    - script: |
+        curl -u "$USERNAME:$PASSWORD" -XPUT "$(ARTIFACTORY-URL)/$(DEPENDENCY-PACK-NAME);deb.distribution=focal;deb.component=main;deb.architecture=amd64;deb.architecture=64bit-arm" -T "output/$(DEPENDENCY-PACK-NAME)" &&
+        curl -u "$USERNAME:$PASSWORD" -XPUT "$(ARTIFACTORY-URL)/$(DEPENDENCY-PACK-PINNING-NAME);deb.distribution=focal;deb.component=main;deb.architecture=amd64;deb.architecture=64bit-arm" -T "output/$(DEPENDENCY-PACK-PINNING-NAME)"
+      displayName: 'Upload packages to Artifactory'
```

Comment:
The 'upload packages to Artifactory' looks like a reusable script across the Dev, Test, Prod stages and can be extracted as an AzDO template. However, we can handle it in a separate PR.

Reply:
I will do this in a later PR